api/middleware/apd.js

```diff
@@ -0,0 +1,108 @@
+const logger = require('../logger')('apd middleware');
+const {
+  apd: ApdModel,
+  apdActivity: defaultActivityModel
+} = require('../db').models;
+
+const cache = {};
+
+module.exports.loadApd = (model = ApdModel, idParam = 'id') => {
+  if (!cache[['loadApd', model, idParam]]) {
+    const loadApd = async (req, res, next) => {
+      logger.silly(req, 'loading APD for request');
+      try {
+        if (model === ApdModel) {
+          req.meta.apd = await ApdModel.where({
+            id: +req.params[idParam]
+          }).fetch({
+            withRelated: ['activities']
+          });
+          next();
+        } else {
+          const obj = await model
+            .where({ id: req.params[idParam] })
+            .fetch({ withRelated: ['apd.activities'] });
+          if (obj) {
+            req.meta.apd = obj.related('apd');
```

Comment:
feels like we should add a log statement here if there's no `obj` instance (or throw an exception) vs. just the silent `next()` with a null `req.meta.apd`

Reply:
Hmm...  Definitely should at least be a log.  Not sure about the throw - that'd cause a 500 error with the body of the exception.  The reason for this `loadApd` middleware being separate is, honestly, premature optimization.  My original plan was to send a 404 if the APD didn't exist (like what `requireApd` does), but then I thought, what if I want to load the APD if it exists, but it's okay if it doesn't?

So it's reasonable to me if you think we should collapse `loadApd` and `requireApd` into the same middleware.  Right now at least, `loadApd` is ***only*** used by `requireApd`, so it shouldn't have any outward-facing implications.  We could always factor it back out later if we needed to, but I'm not sure why we would...  😛

Reply:
Ditched `requireApd`, made `loadApd` return the 404/500 instead.  👍/👎?

Reply:
👍